challenges/easy/who-likes-it/solutions/aldotele/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+func main() {
+  fmt.Println("Enter people below (blank to quit)")
+  likes := enterLikes([]string{}, nil)
+  fmt.Printf(makeSentence(likes))
+}
+
+func enterLikes(x []string, err error) []string {
+    if err != nil {
+        return x
+    }
+    var person string
+    n, err := fmt.Scanf("%s", &person)
+    if n == 1 {
+        x = append(x, person)
+    }
+    return enterLikes(x, err)
+}
+
+func makeSentence(likes []string) string {
+  if len(likes) == 1 {
```

Comment:
Avoid computing slice length on each branch. You can declare a variable such as `likesCount := len(likes)` and use it for each condition. Also, using `switch/case` here could lead to cleaner code

Reply:
Thanks for the advice, I use switch/case now!